commit 343fa1212da77e00e6f960df30fed9ae2368d5dc
Author: eumagga0x2a <eumagga0x2a@users.noreply.github.com>
Date:   Tue Dec 13 12:01:17 2016 +0100

    [build/plugins] Allow system libass, liba52 and libmad

 avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp |  5 +++++
 avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt | 11 +++++++++--
 avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp |  5 +++++
 avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt | 11 +++++++++--
 avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp        |  4 ++++
 avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt        | 51 +++++++++++++++++++++++++++------------------------
 avidemux_plugins/CMakeLists.txt                              | 31 ++++++++++++++++++++++++++++++-
 7 files changed, 89 insertions(+), 29 deletions(-)

commit 9bddbe65d731040e46fcb30c3006796d81956a49
Author: eumagga0x2a <eumagga0x2a@users.noreply.github.com>
Date:   Tue Dec 13 12:30:11 2016 +0100

    [build/plugins] Allow system libs: Remove forgotten leftover

 avidemux_plugins/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

commit ad3af857db49de41e56088a46d250ff22d027a85
Author: eumagga0x2a <eumagga0x2a@users.noreply.github.com>
Date:   Mon Dec 19 20:40:58 2016 +0100

    [build/plugins] Fix build with external libs (oops...)

 avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt | 1 +
 avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt | 1 +
 avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt        | 1 +
 3 files changed, 3 insertions(+)

diff -uprN a/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp b/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp
--- a/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2016-12-19 21:09:28.417839524 +0100
@@ -18,8 +18,13 @@
 #include "ADM_ad_plugin.h"
 
 extern "C" {
+#ifdef USE_EXTERNAL_LIBA52
+#include "a52dec/a52.h"
+#include "a52dec/mm_accel.h"
+#else
 #include "ADM_liba52/a52.h"
 #include "ADM_liba52/mm_accel.h"
+#endif
 };
 
 #define AC3_HANDLE ((a52_state_t *)ac3_handle)
diff -uprN a/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt b/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt
--- a/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2016-12-19 21:09:44.506723455 +0100
@@ -1,11 +1,19 @@
 INCLUDE(ad_plugin)
 
-ADD_SUBDIRECTORY(ADM_liba52)
+IF(NOT USE_EXTERNAL_LIBA52)
+    ADD_SUBDIRECTORY(ADM_liba52)
+ENDIF()
 
 SET(ADM_ad_a52_SRCS ADM_ad_a52.cpp)
 
 ADD_AUDIO_DECODER( ADM_ad_a52  ${ADM_ad_a52_SRCS})
-TARGET_LINK_LIBRARIES(ADM_ad_a52 ADM_liba52)
+
+IF(USE_EXTERNAL_LIBA52)
+    ADD_DEFINITIONS("-DUSE_EXTERNAL_LIBA52")
+    TARGET_LINK_LIBRARIES(ADM_ad_a52 ${LIBA52_LIBRARIES})
+ELSE()
+    TARGET_LINK_LIBRARIES(ADM_ad_a52 ADM_liba52)
+ENDIF()
 
 INIT_AUDIO_PLUGIN(ADM_ad_a52)
 INSTALL_AUDIODECODER(ADM_ad_a52)
diff -uprN a/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp b/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp
--- a/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/ADM_ad_mad.cpp	2016-12-19 21:09:28.418839517 +0100
@@ -16,7 +16,12 @@
  ***************************************************************************/
 #include "ADM_default.h"
 #include "ADM_ad_plugin.h"
+
+#ifdef USE_EXTERNAL_LIBMAD
+#include "mad.h"
+#else
 #include "ADM_libMad/mad.h"
+#endif
 
 #define Stream ((mad_stream *)_stream)
 #define Frame ((mad_frame *)_frame)
diff -uprN a/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt b/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt
--- a/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_audioDecoders/ADM_ad_mad/CMakeLists.txt	2016-12-19 21:09:44.507723448 +0100
@@ -10,13 +10,21 @@ ELSEIF (ADM_CPU_ARMEL)
 	ADD_DEFINITIONS("-DFPM_ARM")
 ENDIF (ADM_CPU_X86_32)
 
-ADD_SUBDIRECTORY(ADM_libMad)
+IF(NOT USE_EXTERNAL_LIBMAD)
+    ADD_SUBDIRECTORY(ADM_libMad)
+ENDIF()
 
 ADD_DEFINITIONS("-DHAVE_ASSERT_H")
 SET(ADM_ad_Mad_SRCS ADM_ad_mad.cpp)
 
 ADD_AUDIO_DECODER( ADM_ad_Mad ${ADM_ad_Mad_SRCS})
-TARGET_LINK_LIBRARIES(ADM_ad_Mad ADM_libMad)
+
+IF(USE_EXTERNAL_LIBMAD)
+    ADD_DEFINITIONS("-DUSE_EXTERNAL_LIBMAD")
+    TARGET_LINK_LIBRARIES(ADM_ad_Mad ${LIBMAD_LIBRARIES})
+ELSE()
+    TARGET_LINK_LIBRARIES(ADM_ad_Mad ADM_libMad)
+ENDIF()
 
 INIT_AUDIO_PLUGIN(ADM_ad_Mad)
 INSTALL_AUDIODECODER(ADM_ad_Mad)
diff -uprN a/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp b/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp
--- a/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_videoFilters6/ass/ADM_vidASS.cpp	2016-12-19 21:09:28.418839517 +0100
@@ -24,7 +24,11 @@
 #include "prefs.h"
 extern "C"
 {
+#ifdef USE_EXTERNAL_LIBASS
+#include "ass/ass.h"
+#else
 #include "ADM_libass/ass.h"
+#endif
 }
 
 /**
diff -uprN a/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt b/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt
--- a/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt	2016-12-19 21:09:44.507723448 +0100
@@ -1,24 +1,28 @@
-INCLUDE(admCheckFreeType)
-INCLUDE(admCheckFridibi)
-ADD_CORE_INCLUDE(ADM_coreSubtitles)
-checkFreeType()
-checkFridibi("0.19")
-
-IF (USE_FREETYPE AND USE_FRIDIBI)
-	ADD_SUBDIRECTORY(ADM_libass)
-
-	INCLUDE(vf_plugin)
-	SET(ADM_vf_ssa_SRCS ADM_vidASS.cpp)
-
-	ADD_VIDEO_FILTER(ADM_vf_ssa ${ADM_vf_ssa_SRCS})
-
-	IF(DO_COMMON)
-		TARGET_LINK_LIBRARIES(ADM_vf_ssa ADM_libass ${FREETYPE2_LDFLAGS} ${FRIDIBI_LDFLAGS} ADM_coreSubtitle)
-                # We force the use of fontconfig
-                ADD_DEFINITIONS("-DUSE_FONTCONFIG")
-                TARGET_LINK_LIBRARIES(ADM_vf_ssa ${FONTCONFIG_LDFLAGS})
-	ENDIF(DO_COMMON)
-
-	INIT_VIDEO_FILTER(ADM_vf_ssa)
-	INSTALL_VIDEO_FILTER(ADM_vf_ssa)
-ENDIF (USE_FREETYPE AND USE_FRIDIBI)
+INCLUDE(admCheckFreeType)
+INCLUDE(admCheckFridibi)
+ADD_CORE_INCLUDE(ADM_coreSubtitles)
+checkFreeType()
+checkFridibi("0.19")
+
+IF (USE_FREETYPE AND USE_FRIDIBI)
+    INCLUDE(vf_plugin)
+    SET(ADM_vf_ssa_SRCS ADM_vidASS.cpp)
+
+    ADD_VIDEO_FILTER(ADM_vf_ssa ${ADM_vf_ssa_SRCS})
+
+    IF(DO_COMMON)
+        IF(USE_EXTERNAL_LIBASS)
+            ADD_DEFINITIONS("-DUSE_EXTERNAL_LIBASS")
+            TARGET_LINK_LIBRARIES(ADM_vf_ssa ${LIBASS_LIBRARIES} ${FREETYPE2_LDFLAGS} ${FRIDIBI_LDFLAGS} ADM_coreSubtitle)
+        ELSE(USE_EXTERNAL_LIBASS)
+            ADD_SUBDIRECTORY(ADM_libass)
+            TARGET_LINK_LIBRARIES(ADM_vf_ssa ADM_libass ${FREETYPE2_LDFLAGS} ${FRIDIBI_LDFLAGS} ADM_coreSubtitle)
+        ENDIF(USE_EXTERNAL_LIBASS)
+        # We force the use of fontconfig
+        ADD_DEFINITIONS("-DUSE_FONTCONFIG")
+        TARGET_LINK_LIBRARIES(ADM_vf_ssa ${FONTCONFIG_LDFLAGS})
+    ENDIF(DO_COMMON)
+
+    INIT_VIDEO_FILTER(ADM_vf_ssa)
+    INSTALL_VIDEO_FILTER(ADM_vf_ssa)
+ENDIF (USE_FREETYPE AND USE_FRIDIBI)
diff -uprN a/avidemux_plugins/CMakeLists.txt b/avidemux_plugins/CMakeLists.txt
--- a/avidemux_plugins/CMakeLists.txt	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_plugins/CMakeLists.txt	2016-12-19 21:09:36.103784041 +0100
@@ -16,7 +16,7 @@ MESSAGE(STATUS "Checking for avidemux de
 IF(NOT FAKEROOT)
 	SET(AVIDEMUX_FAKEROOT "")
 else(NOT FAKEROOT)
-	SET(AVIDEMUX_FAKEROOT "${FAKEROOT}/")
+	SET(AVIDEMUX_FAKEROOT "${FAKEROOT}")
 endif(NOT FAKEROOT)
 
 
@@ -64,6 +64,34 @@ include(admCoreIncludes)
 LINK_DIRECTORIES("${AVIDEMUX_SEARCH_LIB_DIR}")
 INCLUDE_DIRECTORIES("${AVIDEMUX_SEARCH_INCLUDE_DIR}/avidemux/2.6")
 
+#########################################
+# Unbundle some libraries here
+#########################################
+OPTION(USE_EXTERNAL_LIBASS "Use system installed libass library." OFF)
+OPTION(USE_EXTERNAL_LIBMAD "Use system installed libmad library." OFF)
+OPTION(USE_EXTERNAL_LIBA52 "Use system installed liba52 library." OFF)
+
+INCLUDE(FindPkgConfig)
+
+# libass
+IF(USE_EXTERNAL_LIBASS)
+        PKG_CHECK_MODULES(LIBASS REQUIRED libass)
+        INCLUDE_DIRECTORIES(SYSTEM ${LIBASS_INCLUDE_DIRS})
+ENDIF()
+
+# libmad
+IF(USE_EXTERNAL_LIBMAD)
+        FIND_PATH(LIBMAD_INCLUDE_DIR mad.h)
+        FIND_LIBRARY(LIBMAD_LIBRARIES mad)
+        INCLUDE_DIRECTORIES(SYSTEM ${LIBMAD_INCLUDE_DIRS})
+ENDIF()
+
+# liba52
+IF(USE_EXTERNAL_LIBA52)
+        FIND_PATH(LIBA52_INCLUDE_DIR a52dec/a52.h)
+        FIND_LIBRARY(LIBA52_LIBRARIES a52)
+        INCLUDE_DIRECTORIES(SYSTEM ${LIBA52_INCLUDE_DIR})
+ENDIF()
 
 IF (FRESH_BUILD)
 	MESSAGE("")
